test/core/linter_test.py

```diff
@@ -585,3 +589,15 @@ def test_delayed_exception():
     de = runner.DelayedException(ve)
     with pytest.raises(ValueError):
         de.reraise()
+
+
+def test__attempt_to_change_templater_warning(caplog):
+    """Test warning if user tries to change templater in .sqlfluff file in subdirectory."""
+    initial_config = FluffConfig(configs={"core": {"templater": "jinja"}})
+    lntr = Linter(config=initial_config)
+    updated_config = FluffConfig(configs={"core": {"templater": "dbt"}})
+    with caplog.at_level(logging.WARNING):
+        lntr.render_string(
+            in_str="select * from table", fname="test.sql", config=updated_config
+        )
+    assert f"Attempt to set templater to {updated_config.get('templater_obj').name} failed. in caplog.text"
```

Comment:
I think here you meant to have the `in caplog.text` _outside_ the quotes, rather than just asserting that a non-empty string is truthy?

Reply:
woops. Thanks for spotting that @alanmcruickshank